wfl/generate/minimahopping.py

```diff
@@ -1,107 +1,107 @@
-            sys.stderr.flush()
-            os.chdir(workdir)
-            shutil.rmtree(rundir)
-            return None
-        else:
-            raise
-    else:
-        traj = []
-        for hop_traj in Trajectory('minima.traj'):
-            config_type_append(hop_traj, 'hopping_traj')
-            traj.append(hop_traj)
-        os.chdir(workdir)
-        shutil.rmtree(rundir)
-        return traj
-
-
-def _run_autopara_wrappable(atoms, calculator, Ediff0=1, T0=1000, minima_threshold=0.5, mdmin=2,
-                           fmax=1, timestep=1, totalsteps=10, skip_failures=True,
-                           autopara_rng_seed=None, autopara_per_item_info=None,
-                           **opt_kwargs):
-    """runs a structure optimization
-
-    Parameters
-    ----------
-    atoms: list(Atoms)
-        input configs
-    calculator: Calculator / (initializer, args, kwargs)
-        ASE calculator or routine to call to create calculator
-    Ediff0: float, default 1 (eV)
-        initial energy acceptance threshold
-    T0: float, default 1000 (K)
-        initial MD ‘temperature’
-    minima_threshold: float, default 0.5 (Å)
-        threshold for identical configs
-    mdmin: int, default 2
-        criteria to stop MD simulation (number of minima)
-    fmax: float, default 1 (eV/Å)
-        max force for optimizations
-    timestep: float, default 1 (fs)
-        timestep for MD simulations
-    totalsteps: int, default 10
-        number of steps
-    skip_failures: bool, default True
-        just skip optimizations that raise an exception
-    opt_kwargs
-        keyword arguments for MinimaHopping
-    autopara_rng_seed: int, default None
-        global seed used to initialize rng so that each operation uses a different but
-        deterministic local seed, use a random value if None
-
-    Returns
-    -------
-        list(Atoms) trajectories
-    """
-
-    calculator = construct_calculator_picklesafe(calculator)
-    all_trajs = []
-
-    for at_i, at in enumerate(atoms_to_list(atoms)):
-        if autopara_per_item_info is not None:
-            np.random.seed(autopara_per_item_info[at_i]["rng_seed"])
-
-        traj = _atom_opt_hopping(atom=at, calculator=calculator, Ediff0=Ediff0, T0=T0, minima_threshold=minima_threshold,
-                                 mdmin=mdmin, fmax=fmax, timestep=timestep, totalsteps=totalsteps,
-                                 skip_failures=skip_failures, **opt_kwargs)
-        all_trajs.append(traj)
-
-    return all_trajs
-
-
-# run that operation on ConfigSet, for multiprocessing
-def minimahopping(*args, **kwargs):
-    default_autopara_info = {"num_inputs_per_python_subprocess": 10}
-
-    return autoparallelize(_run_autopara_wrappable, *args,
-                           default_autopara_info=default_autopara_info, **kwargs)
-autoparallelize_docstring(minimahopping, _run_autopara_wrappable, "Atoms")
+import os
```

Comment:
What's the final decision about minimahopping? Is this just back to some old version,  or did you decide to implement it without sharing a single minima file between the different autoparallelized tasks?

Reply:
I'll rather just drop parallel minima hopping part but I'd like to update one  thing if possible. In old wfl implementation, minima hopping is ran in tmp folder and erased right after running. Also it returns only minima structures.

But I think it would be useful to keep the trajectories and return MD structures as well for iterative MLP training. So if you think this would be useful as well, I'll update only this feature and push again.

Reply:
That'd be fine by me. You can also take a look at how the DFT calculators do the tmp run directory cleanup, if you want to see how we do it elsewhere in wfl.

Reply:
I have added arguments to `minimahopping.py` that includes md trajectory along with local minima in the output.
Also there are some corresponding modifications in `test_minimahopping.py` as well.